lib/active_merchant/billing/gateways/realex.rb

```diff
@@ -43,6 +43,7 @@ class RealexGateway < Gateway
       def initialize(options = {})
         requires!(options, :login, :password)
         options[:refund_hash] = Digest::SHA1.hexdigest(options[:rebate_secret]) if options.has_key?(:rebate_secret)
+        options[:credit_hash] = Digest::SHA1.hexdigest(options[:credit_secret]) if options.has_key?(:credit_secret)
```

Comment:
I've identified what I believe is a complication here. The user has to pass these values in as credentials in the Spreedly API call (see https://docs.spreedly.com/payment-gateways/realex/). In core, this corresponds to the code here: https://github.com/spreedly/core/blob/master/app/models/gateways/realex_gateway.rb#L10 and here: https://github.com/spreedly/core/blob/master/app/models/gateways/realex_gateway.rb#L18. There, ```refund_password``` is setup as the field that the user passes in to Spreedly, and that field gets converted into the ```rebate_secret``` credential. This seems to be incorrect, because the refund password and rebate password are separate credentials with Realex. In other words, we're calling the Spreedly "rebate" password a "refund" password, when it should really just be the "rebate" password.

So, the issue is what to call the new parameter that the user passes into Spreedly. If not for the existing implementation, I think it would be called ```refund_password```, since that's the name of the field with the user's Realex account. Since that name has been taken, one idea is to call it ```credit_password```, giving us something in core like:

```gateway_field :credit_password, :crypted => true```
```mode.credential :credit_secret, field: :credit_password```

So then the Spreedly user would pass the rebate password into ```<refund_password>``` (which it looks like they must already be doing), and the refund password into ```<credit_password>```. Not ideal, but would be documented, and would avoid breaking current implementations.

Can you please take a look to check my logic, and see if you have any other ideas?

Reply:
Slack conversation: https://spreedly.slack.com/archives/CJ2MQCAJY/p1563808343004700

You seem to be getting a little entangled in the terminology, so I've tried to break it down a bit more here:
Spreedly term: refund  == Realex term: rebate
  refund_password (in Core) == rebate_secret (in Active Merchant) == Rebate Password (in the Realex docs)
Spreedly term: general_credit == Realex term: credit
  credit_password (in Core) == refund_secret (in Active Merchant) == Refund Password (in the Realex docs)
I was previously using credit_secret in Active Merchant, which I have now changed to refund_secret since that seems to paradoxically be less confusing.